pkg/util/errors/k8sutil/k8sutil.go

```diff
@@ -0,0 +1,39 @@
+package k8sutil
+
+import (
+	"github.com/rancher/opni/pkg/util/errors"
+	"google.golang.org/grpc/codes"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func GRPCFromK8s(err error) *errors.GRPCError {
+	switch k8serrors.ReasonForError(err) {
+	case metav1.StatusReasonNotFound, metav1.StatusReasonGone:
+		return errors.New(err, codes.NotFound)
+	case metav1.StatusReasonUnauthorized:
+		return errors.New(err, codes.Unauthenticated)
+	case metav1.StatusReasonForbidden:
+		return errors.New(err, codes.PermissionDenied)
+	case metav1.StatusReasonAlreadyExists:
+		return errors.New(err, codes.AlreadyExists)
+	case metav1.StatusReasonInvalid:
+		return errors.New(err, codes.InvalidArgument)
+	case metav1.StatusReasonServerTimeout, metav1.StatusReasonTooManyRequests:
+		return errors.New(err, codes.ResourceExhausted)
+	case metav1.StatusReasonTimeout:
+		return errors.New(err, codes.DeadlineExceeded)
+	case metav1.StatusReasonBadRequest, metav1.StatusReasonMethodNotAllowed, metav1.StatusReasonConflict:
+		return errors.New(err, codes.FailedPrecondition)
+	case metav1.StatusReasonNotAcceptable, metav1.StatusReasonUnsupportedMediaType:
+		return errors.New(err, codes.Unimplemented)
+	case metav1.StatusReasonInternalError:
+		return errors.New(err, codes.Internal)
+	case metav1.StatusReasonExpired:
+		return errors.New(err, codes.Aborted)
```

Comment:
From the k8s docs:
```
// StatusReasonExpired indicates that the request is invalid because the content you are requesting
	// has expired and is no longer available. It is typically associated with watches that can't be
	// serviced.
	// Status code 410 (gone)
```
To me this aligns with the advice on Aborted:
```
//  (b) Use Aborted if the client should retry at a higher-level
	//      (e.g., restarting a read-modify-write sequence).
```
But I can see the argument for DeadLine Exceeded as well